common/src/main/java/org/conscrypt/OpenSSLCipher.java

```diff
@@ -1010,12 +1029,32 @@ boolean supportsVariableSizeKey() {
         /**
          * The length of the AEAD cipher tag in bytes.
          */
-        private int tagLengthInBytes;
+        int tagLengthInBytes;
 
         public EVP_AEAD(Mode mode) {
             super(mode, Padding.NOPADDING);
         }
 
+        private void checkInitialization() {
+            if (mustInitialize) {
+                throw new IllegalStateException(
+                        "Cannot re-use same key and IV for multiple encryptions");
+            }
+        }
+
+        /** Constant time array comparison. */
+        private boolean arraysAreEqual(byte[] a, byte[] b) {
```

Comment:
I think the purpose is not efficiency but to have a constant-time array equality check.  Arrays.equals() returns as soon as it finds unequal elements, so using it could theoretically open us up to a timing attack (since this is being used to compare keys).  It's probably not a practical attack, but probably worth being a little paranoid.

Reply:
Ah I see ... makes sense.  Maybe add that to the method docs so I don't ask this question again a month from now? :)

Reply:
Yep, done.